lib/mix/lib/mix/project_stack.ex

```diff
@@ -105,7 +105,6 @@ defmodule Mix.ProjectStack do
     end)
   end
 
-  @spec compile_env([term]) :: [term]
```

Comment:
I am not sure what the correct spec would be here, but this function is getting called with `:unset` in `Mix.Tasks.Compile.App.load_compile_env/1`

Reply:
Please open up an issue. :)